File: CS/Using Subreports/XtraReport_Master.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace dxSample {
    public partial class XtraReport_Master : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport_Master() {
            InitializeComponent();
        }

        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XtraReport_Detail)((XRSubreport)sender).ReportSource).Parameters["paramCategoryID"].Value = Convert.ToInt32(GetCurrentColumnValue("CategoryID"));
        }
    }
}

File: VB/Using Subreports/XtraReport_Master.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace dxSample
    Partial Public Class XtraReport_Master
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrSubreport1.BeforePrint
            CType(DirectCast(sender, XRSubreport).ReportSource, XtraReport_Detail).Parameters("paramCategoryID").Value = Convert.ToInt32(GetCurrentColumnValue("CategoryID"))
        End Sub
    End Class
End Namespace
